### PR TITLE
added contact page, nav item, and form API

### LINKED ISSUE #21: NextJS forms, implement basic form
https://nextjs.org/docs/guides/building-forms
Don't need to implement an API, but go as far far down the stack (from the FEnd) as possible

### PR DESCRIPTION
**Pull request by zerske**
resolves #21

### CODE REVIEW

**Comment by yamori** [comment]
```
curl -X POST \
  http://localhost:3000/api/form \
  -H 'Cache-Control: no-cache' \
  -H 'Content-Type: application/json' \
  -H 'Postman-Token: 9b019913-db44-4319-ba8b-276e3fe68a87' \
  -d ' {
  "first": "Kindzerske",
  "last": "jkflds",
  "email": "blah@gmail.com"
}'
```

cool, can hit the api with `curl`

**Comment by zerske** [comment]
Ah durn, should have put “Kindzerske” in the last name field. 😉🤐

File: pages/api/form.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  //   return res.status(400).json({ data: 'First or last name not found' })
+  // }
+
+  switch(true) {
+    case !body.first:
+    case !body.last:
+    case !body.email:
+      return res.status(400).json({ data: 'Email, first name, or last name not found'})
+      break
+    case body.last.toLowerCase() === "kindzerske":
```
Comment (yamori) [inline pages/api/form.js:22]: how about `case /^[kK][a-z]{8}[e]$/.test(body.last):`

Comment (zerske) [reply pages/api/form.js:22]: Tidy bit of regex! 

I see a few reasons to stick with `toLowerCase`. That particular regex wouldn't match `KINDZERSKE` as an input, and if the input is `kindzerske` that regex match is a six step decision tree with roughly half-a-second time-to-match. From my quick and dirty test (and also some recommendations I've read on the internet) using either `toUpperCase`, `localComapre`, or `toLowerCase` is a faster option in Javascript, with `toLowerCase` being the fastest of the three. 

```javascript
// Quick and dirty test

let iterations = 1000;
let data = "kindzerske";
console.time('Function #1');
for(var i = 0; i < iterations; i++ ){
  data.toLowerCase() === "kindzerske";
};
console.timeEnd('Function #1')

console.time('Function #2');
for(var i = 0; i < iterations; i++ ){
    /^[kK][a-z]{8}[e]$/.test(data);
};
console.timeEnd('Function #2')
```

Apparently the best practice is to reserve regex in Javascript for complex matches, defaulting to these simpler methods when applicable.

Comment (yamori) [reply pages/api/form.js:22]: I see, my regex was more about I don't want our last name committed to public code, obfuscate it just a bit.

Good testing above, when you do something like that, also include the CLI output as evidence

**Comment by zerske** [comment]
- [ ] Handle the HTTP response in the front-end, show end-user some sort of 'success' or 'failure'.
- [ ] Write up a test to hit the form's endpoint, check for proper http-400 and -200 responses.

**Comment by zerske** [comment]
I don't think I'll use this in this specific example, but found this write-up interesting and wanted to leave a note. https://www.propelauth.com/post/creating-advanced-reusable-forms-in-next-js

**Pull request closed without merging**